google/cloud/bigtable/internal/data_connection_impl.cc

```diff
@@ -834,7 +835,8 @@ future<StatusOr<bigtable::PreparedQuery>> DataConnectionImpl::AsyncPrepareQuery(
           return std::move(response).status();
         }
 
-        auto refresh_fn = [this, request, current, func]() mutable {
+        auto refresh_fn = [this, request, func]() mutable {
+          auto current = google::cloud::internal::SaveCurrentOptions();
```

Comment:
Mostly documenting this for future reference: does this ensure that the options are preserved/available for the refresh query plan operation(which is performed on a different thread)?

Reply:
This just ensures that when the refresh_fn is called it uses the Options from the current thread. The save/restore Options mechanism is part of the .then lambda on the refresh deadline timer in query_plan.cc